devdoc/UserBuildAssets.md

```diff
@@ -0,0 +1,144 @@
+
+`openssl genrsa -out server.key 2048`
+
+3. generate a csr.conf file
+```
+cat > csr.conf <<EOF
+[ req ]
+default_bits = 2048
+prompt = no
+default_md = sha256
+req_extensions = req_ext
+distinguished_name = dn
+
+[ dn ]
+C = US
+ST = Florida
+L = Boca Raton
+O = LexisNexis Risk
+OU = HPCCSystems Development
+CN = example.com
+
+[ req_ext ]
+subjectAltName = @alt_names
+
+[ alt_names ]
+DNS.1 = example.com
+IP.1 = 127.0.0.1
+
+EOF
+```
+4. Generate Cert Signing Request
+
+`openssl req -new -key server.key -out server.csr -config csr.conf`
+
+5. Create cert.conf file
+```
+cat > cert.conf <<EOF
+
+authorityKeyIdentifier=keyid,issuer
+basicConstraints=CA:FALSE
+keyUsage = digitalSignature, nonRepudiation, keyEncipherment, dataEncipherment
+subjectAltName = @alt_names
+
+[ alt_names ]
+DNS.1 = example.com
+
+EOF
+```
+6. Generate SSL cert with self signed CA
+
+`openssl x509 -req -in server.csr -CA rootCA.crt -CAkey rootCA.key -CAcreateserial -out server.crt -days 365 -sha256 -extfile cert.conf`
+
+7. Use server.crt to generate PKCS12 needed for windows tools
+
+`openssl pkcs12 -inkey server.key -in server.crt -export -name "hpcc_sign_cert" -out hpcc_sign_cert.pfx`
+
+You will be asked to "enter export password", this will be what goes in the variable SIGNING_CERTIFICATE_PASSPHRASE in Github Actions.
+
+8. Convert to base64
+
+On linux:
+`base64 hpcc_sign_cert.pfx > hpcc_sign_cert.base64`
+
+On MacOS:
+`base64 -i hpcc_sign_cert.pfx -o hpcc_sign_cert.base64`
+
+From here you can `cat` the output of hpcc_sign_cert.base64 and copy the output into the variable SIGNING_CERTIFICATE in Github Actions.
+
+### Generating a signing key for linux builds
+For linux builds we're going to generate a private key using GnuPG (gpg).
+
+Start the process by entering a terminal and run the command;`gpg --full-generate-key`
+
+You will be given several options in this process. 
+
+For type of key, select `RSA and RSA default`. 
+
+For keysize, enter `4096`.
+
+For expiration date, select `0 = key does not expire`.
+
+Input your real name.
+
+Input your company email address.
+
+For comment, input something like `Github actions key for signing linux builds`.
+
+Then it will ask you to enter a passphrase for the key, and confirm the passphrase.  Do not leave this blank.
+
+A key should be output and entered into your gpg keychain.  Now we need to export the key for use in the github actions secret.
+
+To extract your key run `gpg --output private.pgp --armor --export-secret-key <email-address-used>`.
+
+Now open private.pgp, copy all, and go to github actions secrets. Paste the output into the secret "SIGNING_SECRET"
+
+## Starting a build
+The build-asset workflow is kicked off by a tag being pushed to the developers HPCC-Platform repository.  Before we push the tag to our HPCC-Platform repository, we will want to have other tags in place if we want LN and ECLIDE builds to function correctly.  Suggested tag patterns are `community_HPCC-12345-rc1` or `HPCC-12345-rc1`.
```

Comment:
Please elaborate on this point. From my own experience, I was not concerned about having either LN or eclide build at all. It looks like the macOS and Windows jobs might combine platform and LN builds in the same job, causing those jobs to replace successful platform builds results with LN errors. A blurb about what to expect if you don't keep all of the repos in sync could make be helpful.